node/src/components/consensus.rs

```diff
@@ -118,6 +118,8 @@ pub enum Event<I> {
     /// Event raised upon initialization, when a number of eras have to be instantiated at once.
     InitializeEras {
         key_blocks: HashMap<EraId, BlockHeader>,
+        /// This is empty except if the activation era still needs to be instantiated: Its
+        /// validator set is read from the global state, not from a key block.
```

Comment:
We could create our own newtype wrapper for this:
```rust
struct<K, V> NonEmptyMap(BTreeMap<K, V>);

impl<K, V> NonEmptyMap {
  fn new(map: BTreeMap) -> Result<Self, Error> {
    // Check if it's non-empty, otherwise return error
  }

  fn new_unsafe(map: BTreeMap) -> Self {
     Self::new(map).expect("input map can't be empty")
  }
```

Reply:
So now you _do_ want to use `Option` again, and the new nonempty map type?

Reply:
Ah no, I don't want to use `Option`. I just thought about what you said earlier about wanting to have a `NonEmptyMap` type. I don't think `Option<Map>` is the same.

Reply:
No, as the comment says, this _is_ empty (formerly `None`) unless we needed to read the validator set from the global state because we can't take it from a key block (i.e. in the genesis/upgrade/restart case).